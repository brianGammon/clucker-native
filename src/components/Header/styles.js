import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  goBackText: {
    paddingLeft: 8,
  },
});
