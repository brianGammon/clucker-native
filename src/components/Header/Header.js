/* @flow */
import React from 'react';
import { withNavigation } from 'react-navigation';
import {
  Header as NBHeader,
  Left,
  Right,
  Body,
  Title,
  Subtitle,
  Button,
  Text,
  Icon,
} from 'native-base';
import { type Navigation } from '../../types';
import ActionButton from '../ActionButton';
import styles from './styles';

type Props = {
  navigation: Navigation,
  title: string,
  subTitle: string | null,
  goBackButton: string,
  cancelButton: boolean,
  eggButton: boolean,
};
export const Header = ({
  navigation,
  title,
  subTitle,
  goBackButton,
  eggButton,
  cancelButton,
}: Props) => (
  <NBHeader>
    <Left>
      {goBackButton !== undefined && (
      <Button transparent onPress={() => navigation.goBack()}>
        <Icon active name="arrow-back" />
        {goBackButton !== '' && (
        <Text style={styles.goBackText}>{goBackButton}</Text>
        )}
      </Button>
      )}
    </Left>
    <Body>
      <Title>{title || 'Default Title'}</Title>
      {subTitle
          && subTitle !== '' && (
            <Subtitle numberOfLines={2} ellipsizeMode="tail">
              {subTitle}
            </Subtitle>
      )}
    </Body>
    <Right>
      {cancelButton && (
      <Button transparent onPress={() => navigation.goBack()}>
        <Text>Cancel</Text>
      </Button>
      )}
      {eggButton && <ActionButton navigation={navigation} />}
    </Right>
  </NBHeader>
);

export default withNavigation(Header);
