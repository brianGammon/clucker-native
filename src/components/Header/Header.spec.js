import React from 'react';
import { Header } from './Header';

describe('Header', () => {
  test('renders with default title, eggButton and goBack button', () => {
    const wrapper = shallow(<Header eggButton goBackButton="Go Back" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders with title set, and cancel button', () => {
    const wrapper = shallow(<Header title="Test Title" cancelButton />);
    expect(wrapper).toMatchSnapshot();
  });
});
