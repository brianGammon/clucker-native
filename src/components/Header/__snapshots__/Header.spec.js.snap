// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders with default title, eggButton and goBack button 1`] = `
<Styled(Header)>
  <Styled(Left)>
    <Styled(Button)
      onPress={[Function]}
      transparent={true}
    >
      <Styled(Icon)
        active={true}
        name="arrow-back"
      />
      <Styled(Text)
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        Go Back
      </Styled(Text)>
    </Styled(Button)>
  </Styled(Left)>
  <Styled(Body)>
    <Styled(Title)>
      Default Title
    </Styled(Title)>
  </Styled(Body)>
  <Styled(Right)>
    <ActionButton
      navigation={
        Object {
          "getParam": [MockFunction],
        }
      }
    />
  </Styled(Right)>
</Styled(Header)>
`;

exports[`Header renders with title set, and cancel button 1`] = `
<Styled(Header)>
  <Styled(Left) />
  <Styled(Body)>
    <Styled(Title)>
      Test Title
    </Styled(Title)>
  </Styled(Body)>
  <Styled(Right)>
    <Styled(Button)
      onPress={[Function]}
      transparent={true}
    >
      <Styled(Text)>
        Cancel
      </Styled(Text)>
    </Styled(Button)>
  </Styled(Right)>
</Styled(Header)>
`;
